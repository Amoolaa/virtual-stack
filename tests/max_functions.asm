FUNC LABEL 0
    MOV REG 0 VAL 9
    MOV REG 1 VAL 254
    PRINT REG 0
    CAL VAL 2
    RET
FUNC LABEL 2
    ADD REG 0 REG 1
    CAL VAL 3
    CAL VAL 1
    RET
FUNC LABEL 1
    ADD REG 0 REG 1
    RET
FUNC LABEL 3
    ADD REG 0 REG 1
    PRINT REG 0
    CAL VAL 5
    RET
FUNC LABEL 5
    ADD REG 0 REG 1
    MOV STK A VAL 0
    MOV REG 2 STK A
    REF STK B STK A
    MOV REG 1 PTR B
    EQU REG 1
    PRINT REG 1
    CAL VAL 4
    RET
FUNC LABEL 4
    PRINT REG 2
    CAL VAL 6
    RET
FUNC LABEL 6
    NOT REG 0
    PRINT REG 0
    CAL VAL 7
    RET
FUNC LABEL 7
    PRINT REG 7
    RET
