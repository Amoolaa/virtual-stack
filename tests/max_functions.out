9
5
1
0
252
1
